### ISSUE TITLE
Switch bank transfer payments to use Transferwise

### ISSUE THREAD

**Issue by russss** [emfcamp/Website#857]
Looks like the [Transferwise API](https://api-docs.transferwise.com/) is available for us to use now. It would be good to use Transferwise for all incoming bank payments, as they have a proper API.

We can automatically create the bank accounts in our DB by calling [get account balance](https://api-docs.transferwise.com/#borderless-accounts-get-account-balance) - probably needs to be an admin job to do that.

Then we need to listen for the [balance deposit event](https://api-docs.transferwise.com/#webhook-events-balance-deposit-event) webhook, fire off a [get account statement](https://api-docs.transferwise.com/#borderless-accounts-get-account-statement) request for that bank account, import any un-imported transactions, then reconcile.

This will be (possibly much) faster, remove the dependency on my bank scraping infrastructure, and fix #570.

**Comment by jayaddison** [comment]
Here's my current understanding of the issue - much of this will just be echoing the description to see whether I've understood it correctly.

The plan is to listen for incoming balance credits to a receiving TransferWise bank account, by using a webhook subscription created for the TransferWise profile.  This subscription will listen for `balances#credit` events (the only event type we need to listen for currently).

When a credit is received, we'll request[q1] a recent account statement using the `profileId` from the webhook and `borderlessAccountId` for the account[q2].

For each transaction in the response, we'll find a corresponding `BankPayment` object (if any) based on the `paymentReference`, validate the payment amount and details, and transition the payment into an appropriate state.

Questions:

1. What time interval should the account statements be requested for - and will this be based on a 'high water mark' timestamp of previous successfully processed transactions, or should we just request the last `n` days each time?  (max `n` == 469 days according to the TransferWise docs)
1. Where will the `borderlessAccountId` be modelled in the DB?
1. More generally, how are the relevant bank accounts modelled in the DB, and/or what will be required to enable an admin task for this? (cc @russss)

**Comment by russss** [comment]
> For each transaction in the response, we'll find a corresponding BankPayment object (if any) based on the paymentReference, validate the payment amount and details, and transition the payment into an appropriate state.

There are actually two steps here:

1. Assuming the payment is a credit (we ignore debits entirely), first we insert the transaction as a [BankTransaction](https://github.com/emfcamp/Website/blob/877510cab3626fb83a71bb7a24cb574d891b83a7/models/payment.py#L325), if it doesn't already exist.
2. Then we try and reconcile the transaction with existing BankPayments.

This allows us to manually reconcile payments which have incorrect references/amounts in the admin interface.

You can see how we currently import BankTransactions in the [loadofx task](https://github.com/emfcamp/Website/blob/877510cab3626fb83a71bb7a24cb574d891b83a7/apps/base/tasks_banking.py#L39), and the reconciliation happens in the [reconcile task](https://github.com/emfcamp/Website/blob/877510cab3626fb83a71bb7a24cb574d891b83a7/apps/base/tasks_banking.py#L133). 

We could keep the reconcile task, but I think that's unnecessary - we should just be able to run the reconciliation logic in the webhook. We'll need to keep these running in parallel for a short while but I'd like to remove both these tasks once we're done.

> 1. What time interval should the account statements be requested for - and will this be based on a 'high water mark' timestamp of previous successfully processed transactions, or should we just request the last `n` days each time?  (max `n` == 469 days according to the TransferWise docs)

Given that we should be receiving a webhook for every transaction, in theory we shouldn't need more than a day's worth of transactions. Maybe say a week to be safe. It looks like there's no limit to the number of transactions returned - or at least not one we're likely to hit.

> * Where will the `borderlessAccountId` be modelled in the DB?
> * More generally, how are the relevant bank accounts modelled in the DB, and/or what will be required to enable an admin task for this?

They're stored in the [BankAccount](https://github.com/emfcamp/Website/blob/877510cab3626fb83a71bb7a24cb574d891b83a7/models/payment.py#L299) table, which the BankTransactions are keyed off. So the Borderless ID will need to be a new column in there.

It would be nice to have an admin command that fetches all the Transferwise accounts (for our supported currencies) and inserts them into that table automatically. My slight preference is for this to be done through the admin web interface rather than a CLI command, but I'll leave that up to you.

We can also use the BankAccount table to populate the bank details we show to people on the website, which will neatly solve #570. To do this, it looks like we'll also need IBAN/SWIFTBIC columns in that table as well (and make account/sort code nullable). We'll also need a boolean "active" column to select which account we want to show for a given currency.

**Comment by marksteward** [comment]
I think this is now working - we handle webhooks and sync transactions against the appropriate account statement. Reconciliation is done as a separate job, but it's probably best left until we've got some data.
